ducktape/services/service.py

```diff
@@ -100,11 +102,29 @@ def start(self):
 
         self.logger.debug(self.who_am_i() + ": killing processes and attempting to clean up before starting")
         for node in self.nodes:
-            # Added precaution - kill running processes
-            self._kill_running_processes(node)
-            self.force_clean_node(node)
-            self.stop_node(node)
-            self.clean_node(node)
+            # Added precaution - kill running processes, clean persistent files
+            # try/except for each step, since each of these steps may fail if there are no processes
+            # to kill or no files to remove
+
+            try:
+                self._kill_running_processes(node)
+            except:
+                pass
+
+            try:
+                self.force_clean_node(node)
+            except:
+                pass
+
+            try:
+                self.stop_node(node)
```

Comment:
Hmm, I haven't looked at this code for awhile, but how does this sequence of calls make sense? It doesn't seem like stop_node makes sense if we have something called _kill_running_processes. And force_clean_node seems to delete stuff out of /mnt, but happens before stop_node, but then couldn't something in stop_node log to a file in /mnt?

A couple of other problems:

1. Can we just drop _kill_running_processes and delete that method? It's just killing all java processes, which isn't a good solution in ducktape? If we want to provide something like that for java processes it should be in kafka/muckrake. Plus, it's not even really safe if there are any other java processes that might be legitimately running -- it was just an ugly hack when I could assume this was all in a VM.

2. force_clean_node's implementation is assuming the layout that we're using for kafka/muckrake. We can't really provide that as a default implementation in ducktape. Again, we could move this out to some common Service base class/mixin in kafka/muckrake. Alternatively, both force_clean_node (which handles files) and _kill_running_processes seem like things you might want to specify at the cluster level, so maybe there's a better hook to be provided for them in ducktape. Or yet another alternative would be removing them entirely -- they actually should only be necessary during development since robust Service implementations won't leave any trace if they clean up properly.